src/sentry/sentry_metrics/metrics_interface.py

```diff
@@ -13,9 +14,10 @@ def counter(
         project_id: int,
         metric_name: str,
         value: Union[int, float],
-        unit: Optional[str],
         tags: Mapping[str, str],
-    ) -> None:
+        unit: Optional[str] = None,
```

Comment:
Is the unit usually supposed to be none under normal operation? I would avoid adding this default if not - better to force the user to pass it along than accidentally apply None

Reply:
Looking through the metrics abstraction layer code in Sentry, it looks like the unit is passed in/parsed directly from MRI and treated as separate from the metric name. I think I might want to keep it this way for now, but will also see whether users of the interface prefer something else.

Reply:
i think lyn is suggesting that you make this parameter required, but not sure

Reply:
We had a conversation offline about why it needs to exist at all. It seems to just add unnecessary complexity to me. Still, if we decide we want to have it, then I'd prefer not silently applying defaults.

Reply:
Not sure if this can be a required field since `none` is considered a valid unit. Even if the default is removed and the field is kept Optional, `none` will need to be applied in order to build the correct MRI.

Reply:
Removed the default and made the unit just Optional